week1/homework/src/server.js

```diff
@@ -9,12 +9,36 @@ function createServer(port) {
   let state = 10;
 
   const server = http.createServer((request, response) => {
-    // TODO: Write your homework code here
+    switch (request.url) {
+      case '/add':
+        response.setHeader('Content-Type', 'application/json');
+        state++;
+        response.end(JSON.stringify({ state: state }));
+        break;
+      case '/state':
+        response.setHeader('Content-Type', 'application/json');
+        response.write(JSON.stringify({ state: state }));
+        break;
+      case '/subtract':
+        response.setHeader('Content-Type', 'application/json');
+        state--;
+        response.write(JSON.stringify({ state: state }));
```

Comment:
💡 PRO TIP: If you find yourself repeating a piece of code perhaps it is a good idea to put it in a seperate function instead of copy-pasting it

Reply:
I ve changed it.